mapie/quantile_regression.py

```diff
@@ -0,0 +1,323 @@
+                                "The matching parameter for the quantile"
+                                "value of this method does not exist."
+                            )
+                else:
+                    raise ValueError(
+                        "The value specified as link to the quantile"
+                        "method does not exist."
+                    )
+            else:
+                raise ValueError(
+                    "We cannot find your method to have a link"
+                    "to a quantile method."
+                )
+        return estimator
+
+    def _check_size_calib_set(self, y, alpha) -> None:
+        value = 2/((len(y)/2)+1)
+        if alpha < value:
+            raise ValueError(
+                "The calibration set is too small."
+            )
+
+    def _check_validity_results(
+        self,
+        y_preds: NDArray,
+        y_pred_low: ArrayLike,
+        y_pred_up: ArrayLike
+    ) -> None:
+        """_summary_
+
+        Parameters
+        ----------
+        y_preds : _type_
+            _description_
+        y_pred_low : _type_
+            _description_
+        y_pred_up : _type_
+            _description_
+
+        Raises
+        ------
+        Warning
+            If the aggregated predictions of any training sample would be nan.
+        Examples
+        --------
+        >>> import warnings
+        >>> warnings.filterwarnings("error")
+        >>> import numpy as np
+        >>> from mapie.quantile_regression import _check_validity_results
+        >>> y_preds = np.array([[1, 2, 3],[3, 4, 5],[2, 3, 4]])
+        >>> y_pred_low = np.array([4, 3, 2])
+        >>> y_pred_up = np.array([4, 4, 4])
+        >>> try:
+        ...     _check_validity_results(y_preds, y_pred_low, y_pred_up)
+        ... except Exception as exception:
+        ...     print(exception)
+        ...
+        WARNING: Not correct.
+        """
+        if (
+            (y_pred_low >= y_pred_up).all() or
+                (y_preds[2] <= y_pred_low).all() or
+                (y_preds[2] >= y_pred_up).all()):
+            warnings.warn(
+                "WARNING: Not correct."
+            )
+
+    def _check_cv(
+        self,
+        cv: Optional[Union[int, str, BaseCrossValidator]] = None
+    ) -> Union[str, BaseCrossValidator]:
+        if cv is None:
+            cv == "split"
+            return cv
+        if cv == "split":
+            return cv
+        else:
+            raise ValueError(
+                "Invalid cv method."
+            )
+
+    def fit(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+        sample_weight: Optional[ArrayLike] = None,
+    ) -> MapieQuantileRegressor:
+        # Checks
+        self._check_parameters()
+        cv = self._check_cv(self.cv)
+        estimator = self._check_estimator(self.estimator)
+        alpha = self._check_alpha(self.alpha)
+        self._check_size_calib_set(y, self.alpha)
+        alpha_copy = alpha.copy()
```

Comment:
Why do you need to copy the `alpha` created in `_check_alpha` ?

Reply:
It made sense when we changed what self.alpha was. With the implementation of adding 0.5 directly in the method, it is no longer needed.